repositories/mentor_repository.js

```diff
@@ -19,34 +20,80 @@ async function getMentorByUserId(userId) {
 
 async function getMentorById(allowedStatuses, selectedFields, mentorId) {
   const knex = getKnex();
-  return await knex("mentors")
+  const mentor = await knex("mentors")
     .join("users", "mentors.user_id", "users.id")
     .where("mentors.id", mentorId)
     .whereIn("mentors.status", allowedStatuses)
     .select(selectedFields).first();
+
+  const mentorTags = await tagsRepo.getAssignedTagNames(
+    mentorId,
+    'mentor'
+  );
+
+  mentor.tags = mentorTags;
+  
+  return mentor;
 }
 
 async function createMentor(mentorData) {
   const knex = getKnex();
+  const { tags = [], ...data } = mentorData;
 
-  const [createdMentor] = await knex("mentors")
-    .insert(mentorData)
-    .returning("id");
+  return await knex.transaction(async (trx) => {
+    const [createdMentor] = await trx("mentors")
+      .insert(data)
+      .returning("id");
 
-  const result = await knex("mentors")
-    .join("users", "mentors.user_id", "users.id")
-    .select("mentors.id", "users.name", "mentors.about")
-    .where("mentors.id", createdMentor.id)
-    .first();
+    await updateMentorTags(createdMentor.id, tags, trx);
+
+    const result = await trx("mentors")
+      .join("users", "mentors.user_id", "users.id")
+      .select("mentors.id", "users.name", "mentors.about")
+      .where("mentors.id", createdMentor.id)
+      .first();
 
-  return result;
+    return result;
+  });
+}
+
+async function updateMentorTags(mentorId, tags = [], trx) {
+  const existing = await tagsRepo.getAssignedTagNames(mentorId, 'mentor', trx);
+
+  const toRemove = existing.filter((name) => !tags.includes(name));
+  const toAdd = tags.filter((name) => !existing.includes(name));
+
+  if (toRemove.length) {
+    const idsToRemove = await Promise.all(
+      toRemove.map((name) =>
+        tagsRepo.findTagByName(name, trx).then((t) => t.id)
+      )
+    );
+    await tagsRepo.removeAssignments(idsToRemove, mentorId, 'mentor', trx);
+  }
+
+  for (const name of toAdd) {
+    const tag = await tagsRepo.ensureTag(name, trx);
+    await tagsRepo.assignTagTo(tag.id, mentorId, 'mentor', trx);
```

Comment:
what about using batchInsert instead of separate call per tag? 
https://knexjs.org/guide/utility.html#batchinsert